src/bitmap/ops.rs

```diff
@@ -439,9 +441,247 @@ impl BitXorAssign<&RoaringBitmap> for RoaringBitmap {
     }
 }
 
+impl<I> IterExt<RoaringBitmap> for I
+where
+    I: IntoIterator<Item = RoaringBitmap>,
+{
+    type Bitmap = RoaringBitmap;
+
+    fn or(self) -> Self::Bitmap {
+        try_naive_lazy_multi_op_owned(
+            self.into_iter().map(Ok::<_, std::convert::Infallible>),
+            |a, b| BitOrAssign::bitor_assign(a, b),
+        )
+        .unwrap()
+    }
+
+    fn and(self) -> Self::Bitmap {
+        try_simple_multi_op_owned(
+            self.into_iter().map(Ok::<_, std::convert::Infallible>),
+            |a, b| BitAndAssign::bitand_assign(a, b),
+        )
+        .unwrap()
+    }
+
+    fn sub(self) -> Self::Bitmap {
+        try_simple_multi_op_owned(
+            self.into_iter().map(Ok::<_, std::convert::Infallible>),
+            |a, b| SubAssign::sub_assign(a, b),
+        )
+        .unwrap()
+    }
+
+    fn xor(self) -> Self::Bitmap {
+        try_naive_lazy_multi_op_owned(
+            self.into_iter().map(Ok::<_, std::convert::Infallible>),
+            |a, b| BitXorAssign::bitxor_assign(a, b),
+        )
+        .unwrap()
+    }
+}
+
+impl<I, E> IterExt<Result<RoaringBitmap, E>> for I
+where
+    I: IntoIterator<Item = Result<RoaringBitmap, E>>,
+{
+    type Bitmap = Result<RoaringBitmap, E>;
+
+    fn or(self) -> Self::Bitmap {
+        try_naive_lazy_multi_op_owned(self, |a, b| BitOrAssign::bitor_assign(a, b))
+    }
+
+    fn and(self) -> Self::Bitmap {
+        try_simple_multi_op_owned(self, |a, b| BitAndAssign::bitand_assign(a, b))
+    }
+
+    fn sub(self) -> Self::Bitmap {
+        try_simple_multi_op_owned(self, |a, b| SubAssign::sub_assign(a, b))
+    }
+
+    fn xor(self) -> Self::Bitmap {
+        try_naive_lazy_multi_op_owned(self, |a, b| BitXorAssign::bitxor_assign(a, b))
+    }
+}
+
+impl<'a, I> IterExt<&'a RoaringBitmap> for I
+where
+    I: IntoIterator<Item = &'a RoaringBitmap>,
+{
+    type Bitmap = RoaringBitmap;
```

Comment:
Would it make sense to also impl `impl<I, E> IterExt<Result<&RoaringBitmap, E>> for I`?

Reply:
Oops yes sorry I'm putting this back in draft!
I wanted to benchmark the performances with and without this change before moving forward!